### PR TITLE
(bugfix): zinit self-update uses current branch

### PR DESCRIPTION
**Pull request by vladdoster**
fixes bug where zinit self-update was hard-coded to use main, even if it was on a different branch.

Signed-off-by: Vladislav Doster <mvdoster@gmail.com>

### CODE REVIEW

**Comment by vladdoster** [comment]
image

**Comment by vladdoster** [comment]
carbon

**Comment by vladdoster** [comment]
```zsh
~ bugfix/self-update-can-use-any-branch
ᐳ zi self-update
[self-update] fetching changes for bugfix/self-update-can-use-any-branch
From https://github.com/zdharma-continuum/zinit
 * branch              bugfix/self-update-can-use-any-branch -> FETCH_HEAD
Already up to date.
[self-update] Compiling Zinit (zcompile)…
[self-update] Reloading Zinit for the current session…

~ bugfix/self-update-can-use-any-branch
ᐳ
```

**Comment by alichtman** [comment]
Can you regenerate the docs?

File: zinit-autoload.zsh [hunk 1]
```diff
@@ -711,12 +711,13 @@ ZINIT[EXTENDED_GLOB]=""
...
 
-    [[ $1 = -q ]] && +zinit-message "{info2}Updating Zinit{…}{rst}"
+    [[ $1 = -q ]] && +zinit-message "{pre}[self-update]{msg2}{info2}Updating Zinit{…}{rst}"
 
     local nl=$'\n' escape=$'\x1b['
+    local current_branch=$(command pushd ZINIT[BIN_DIR] && git branch --show-current && popd)
     local -a lines
     (   builtin cd -q "$ZINIT[BIN_DIR]" && \
-        command git checkout main &>/dev/null && \
+        +zinit-message -n "{pre}[self-update]{msg2} fetching changes for {msg2}$current_branch{cmd}" && \
```
Comment (alichtman) [inline zinit-autoload.zsh:720]:
```suggestion
        +zinit-message -n "{pre}[self-update]{msg2} fetching changes from {msg2}$current_branch{cmd}" && \
```

File: zinit-install.zsh [hunk 2]
```diff
@@ -1447,25 +1447,25 @@ builtin source "${ZINIT[BIN_DIR]}/zinit-side.zsh" || {
         local url=https://$urlpart
```
Comment (alichtman) [inline zinit-install.zsh:1447]: Changes in this file seem unrelated. Looks like `bpick` work and work from other PRs. Can you unstage these?

**Review by alichtman — CHANGES_REQUESTED** [review]

File: zinit-install.zsh (hunk 2 above)
Comment (vladdoster) [reply zinit-install.zsh:1447]: Correct. Snuck in from another branch. This code belongs in #201.

I have removed it.

**Comment by vladdoster** [comment]
Screen Shot 2022-04-11 at 04 47 48

**Review by alichtman — APPROVED** [review]

File: zinit-autoload.zsh [hunk 3]
```diff
@@ -711,14 +711,16 @@ ZINIT[EXTENDED_GLOB]=""
...
     local -a lines
-    (   builtin cd -q "$ZINIT[BIN_DIR]" && \
-        command git checkout main &>/dev/null && \
-        command git fetch --quiet && \
-            lines=( ${(f)"$(command git --no-pager log --color --date=short --pretty=format:'%Cgreen%cd %h %Creset%s %Cred%d%Creset || %b' ..FETCH_HEAD)"} )
+    (
+        builtin cd -q "$ZINIT[BIN_DIR]" \
+        && +zinit-message -n "{pre}[self-update]{msg2} fetching latest changes from {msg2}$current_branch$nl{cmd}" \
+        && command git fetch --quiet \
+        && lines=( ${(f)"$(command git log --color --date=short --pretty=format:'%Cgreen%cd %h %Creset%s %Cred%d%Creset || %b' ..FETCH_HEAD)"} )
```
[the remark this answers is not in the document]
Comment (yous) [reply zinit-autoload.zsh:723]: Still it seems that there is no `--no-pager` in `zinit self-update` command. When it will be added?

[the remark this answers is not in the document]
Comment (vladdoster) [reply zinit-autoload.zsh:723]: @yous,

Please open a new issue instead of commenting on a two year old PR.
